### PR TITLE
Ibor Binary Caplet/Floolet, single period.

### PR DESCRIPTION
**Pull request by marc-henrard**
* Trade model for a single period binary caplet/floorlet
* Pricer based on call spread approach

### CODE REVIEW

File: modules/pricer/src/test/java/com/opengamma/strata/pricer/capfloor/VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+      .iborRate(RATE_COMP)
+      .build();
+
+  // valuation date before fixing date
+  private static final ImmutableRatesProvider RATES = IborCapletFloorletSabrRateVolatilityDataSet.getRatesProvider(
+      VALUATION.toLocalDate(), EUR_EURIBOR_3M, LocalDateDoubleTimeSeries.empty());
+  private static final SabrParametersIborCapletFloorletVolatilities VOLS = IborCapletFloorletSabrRateVolatilityDataSet
+      .getVolatilities(VALUATION, EUR_EURIBOR_3M);
+  // valuation date after fixing date
+  // valuation date equal to fixing date
```
Comment (skitini) [inline modules/pricer/src/test/java/com/opengamma/strata/pricer/capfloor/VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest.java:82]: Looks like this comment can be removed as the rates provider below is for val date after fixing date?

File: modules/pricer/src/test/java/com/opengamma/strata/pricer/capfloor/VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+import com.opengamma.strata.market.sensitivity.PointSensitivityBuilder;
+import com.opengamma.strata.pricer.rate.ImmutableRatesProvider;
+import com.opengamma.strata.product.capfloor.IborCapletFloorletBinaryPeriod;
+import com.opengamma.strata.product.capfloor.IborCapletFloorletPeriod;
+import com.opengamma.strata.product.rate.IborRateComputation;
+
+/**
+ * Test {@link VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricer}.
+ */
+public class VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest {
```
Comment (skitini) [inline modules/pricer/src/test/java/com/opengamma/strata/pricer/capfloor/VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest.java:34]: Test class and methods don't need to be public.

File: modules/product/src/test/java/com/opengamma/strata/product/capfloor/IborCapletFloorletBinaryPeriodTest.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.currency.CurrencyAmount;
+import com.opengamma.strata.product.common.PutCall;
+import com.opengamma.strata.product.rate.IborRateComputation;
+
+/**
+ * Test {@link IborCapletFloorletBinaryPeriod}.
+ */
+public class IborCapletFloorletBinaryPeriodTest {
```
Comment (skitini) [inline modules/product/src/test/java/com/opengamma/strata/product/capfloor/IborCapletFloorletBinaryPeriodTest.java:32]: Same here, the classes and methods can be package private

**Review by skitini — CHANGES_REQUESTED** [review]

File: modules/pricer/src/test/java/com/opengamma/strata/pricer/capfloor/VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest.java (hunk 1 above)
Comment (marc-henrard) [reply modules/pricer/src/test/java/com/opengamma/strata/pricer/capfloor/VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest.java:82]: Yes, copy and paste from another test. The fixing date issue is dealt (and tested) with at the "VolatilityIborCapletFloorletPeriodPricer" level. I remove the comments.

File: modules/pricer/src/test/java/com/opengamma/strata/pricer/capfloor/VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest.java (hunk 2 above)
Comment (marc-henrard) [reply modules/pricer/src/test/java/com/opengamma/strata/pricer/capfloor/VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest.java:34]: I copy and paste old tests. I don't know what is the convention now. I remove the "public" markers from the abvoe tests.

**Review by skitini — APPROVED** [review]

**Pull request merged**